### ISSUE TITLE
Cloud Function deployed via Terraform not responding to requests

### ISSUE THREAD

**Issue by dedeco** [GoogleCloudPlatform/bigquery-dlp-remote-function#101]
I'm experiencing issues with a Cloud Function deployed using Terraform. While the Terraform deployment appears successful, attempts to call the function are failing.

**Steps to reproduce:**

1. Deployed the Cloud Function using Terraform (relevant configuration snippets available if needed).
2. Attempted to invoke the function using the following curl command:

```bash
curl -X POST https://bq-transform-fns-1075826705952.us-central1.run.app \
-H "Authorization: bearer $(gcloud auth print-identity-token)" \
-H "Content-Type: application/json" \
-d '{
  "requestId": "deidRequest123", 
  "caller": "//bigquery.googleapis.com/projects/myproject/jobs/myjobid", 
  "sessionUser": "abc@abc.com",
  "userDefinedContext": {
    "mode": "deidentify",
    "algo": "dlp"
  },
  "calls": [
    [
      "Send a note to test@example.com." 
    ],
    [
      "(555) 253-0000"
    ]
  ]
}'
```

**Observed behavior:**

The request fails with the following error: "The request failed because either the HTTP response was malformed or connection to the instance had an error."

**Troubleshooting:**

* I've reviewed the troubleshooting documentation at [https://cloud.google.com/run/docs/troubleshooting#malformed-response-or-connection-error](https://cloud.google.com/run/docs/troubleshooting#malformed-response-or-connection-error), but the suggestions provided haven't resolved the issue.
* I'm happy to provide the specific error message, Cloud Function logs, and relevant Terraform configuration snippets if that would be helpful.

**Additional context:**

* The Cloud Function is intended to be called from BigQuery, but direct invocation via curl is also failing.

I would appreciate any assistance in diagnosing and resolving this problem.

**Comment by anantdamle** [comment]
Thanks for a detailed writeup. Let me investigate the issue later today

**Comment by anantdamle** [comment]
On further investigation, I found that its caused by a JVM core-dump, this requires some more tests. please stay tuned.

**Comment by dedeco** [comment]
I encountered an issue with Java 21 when using the latest `eclipse-temurin:21-jre-alpine` image. My containers were crashing, and after some investigation (including checking out similar issues on GitHub like [https://github.com/adoptium/adoptium-support/issues/1185](https://github.com/adoptium/adoptium-support/issues/1185)), I found that downgrading to `eclipse-temurin:21.0.4_7-jre` resolved the problem.

To implement this fix, I updated my `build.gradle` file to use the specific, older version:

```gradle
jib {
    from {
        image = 'eclipse-temurin:21.0.4_7-jre' 
    }
}
```

This change ensures my builds use a stable Java 21 version that avoids the crashing issue.
